### PR TITLE
LC-597 - additional login checks

### PR DESCRIPTION
**Pull request by m24murray**
Adding new UserDetailsChecker to determine if a user is whitelisted, part of an agency or has been manually invited via IDM. If false, the login will be rejected via an updated CustomerAuthFailureHandler and a new AccountBlockedException will be thrown.

Updating login page to show new account blocker error message.

Updating invite service and repository to check if email has been manually invited via idm.

### CODE REVIEW

File: src/main/java/uk/gov/cshr/repository/InviteRepositoryMockImpl.java [hunk 1]
```diff
@@ -65,6 +65,11 @@ public boolean existsByForEmailAndStatus(String email, InviteStatus status) {
         return false;
     }
 
+    @Override
```
Comment (brdgeth) [inline src/main/java/uk/gov/cshr/repository/InviteRepositoryMockImpl.java:68]: Can this class by removed?  As I added this when I first joined the project to try and mock the GovNotify.  A few months later I remember you saying you deleted all the spring profiles when you did a production release.  Therefore its not being used anymore.

File: src/main/java/uk/gov/cshr/config/CustomAuthenticationFailureHandler.java [hunk 2]
```diff
@@ -11,10 +11,17 @@
 public class CustomAuthenticationFailureHandler implements AuthenticationFailureHandler {
     @Override
     public void onAuthenticationFailure(HttpServletRequest request, HttpServletResponse response, AuthenticationException exception) throws IOException, ServletException {
-        if (exception.getMessage().equals("User account is locked")) {
-            response.sendRedirect("/login?error=locked");
-        } else {
-            response.sendRedirect("/login?error=failed");
+        String exceptionMessage = exception.getMessage();
```
Comment (brdgeth) [inline src/main/java/uk/gov/cshr/config/CustomAuthenticationFailureHandler.java:14]: I am a bit suspicious about this.  Is there a better way of doing this other than using the exception message?
(As it is controlled by Spring).

Comment (mattdoherty-kainos) [reply src/main/java/uk/gov/cshr/config/CustomAuthenticationFailureHandler.java:14]: Might be best to use the type

Comment (m24murray) [reply src/main/java/uk/gov/cshr/config/CustomAuthenticationFailureHandler.java:14]: This customer failure handler class and the `exception.getMessage()` was already in place before my updates for 'account blocked'. I'm going to keep it in for now as this is already quite a significant diff and don't want to rework everything in identity-service.

File: src/main/java/uk/gov/cshr/repository/InviteRepositoryMockImpl.java (hunk 1 above)
Comment (m24murray) [reply src/main/java/uk/gov/cshr/repository/InviteRepositoryMockImpl.java]: Class removed

**Pull request merged**